pkg/apis/application/v1alpha1/types.go

```diff
@@ -1056,15 +1056,15 @@ func (a *ApplicationStatus) GetRevisions() []string {
 
 // BuildComparedToStatus will build a ComparedTo object based on the current
 // Application state.
-func (app *Application) BuildComparedToStatus() ComparedTo {
+func (spec *ApplicationSpec) BuildComparedToStatus() ComparedTo {
```

Comment:
Scoped this down since we don't need access to the whole app. Makes it easier to avoid doing a `DeepCopy` on the whole app in `specEqualsCompareTo`